### ISSUE TITLE
Bug: Filename separator Title - Label Number does not work correctly if there is a hyphen between label and number

### ISSUE THREAD

**Issue by phoef** [mpogue2/SquareDesk#1152]
SquareDesk V1.0.9 (Qt6.7.2), Darkmode

I have some Labels, where I use a hyphen within the "Label Number" section. It seems to be, that SquareDesk uses just the hyphen as separator and "space hyphen space" (" - ")? I added two examples where parts of the Label are placed within the title section. 

<img width="239" alt="Bildschirmfoto 2024-07-29 um 18 30 42" src="https://github.com/user-attachments/assets/64423604-ce3c-4e5c-a952-058c8fa521e4"><br>
<img width="564" alt="Bildschirmfoto 2024-07-29 um 18 26 43" src="https://github.com/user-attachments/assets/fb292b54-278a-4dbf-8dd9-b0ccd6d7d6ac">

<img width="274" alt="Bildschirmfoto 2024-07-29 um 18 30 59" src="https://github.com/user-attachments/assets/76062ec8-2d69-4481-8f4d-a2cf846d9b62">
<br><img width="613" alt="Bildschirmfoto 2024-07-29 um 18 28 24" src="https://github.com/user-attachments/assets/b05956a1-62b1-40f0-bd88-3f1d76cc1e12">

**Comment by mpogue2** [comment]
Thanks for the detailed bug report!

Label names with dashes in them often must be hand-coded, because the dashes make splitting the title into pieces ambiguous (and we will often pick the wrong one).  This is usually a fairly straightforward change, and I'll try to get this done soon.

**Comment by mpogue2** [comment]
@phoef I was able to replicate the bug with Circle-D, but as you can see below, V1.0.9 in DarkMode appears to handle the 4-Bar-B case correctly.

Maybe could you cut/paste the exact title of the 4-Bar-B song into this Issue (as text, rather than a picture).  Maybe one of the dashes is a weird dash that _looks_ like a dash, but isn't? :-)

In Finder:
![image](https://github.com/user-attachments/assets/d09323c9-460b-436e-b0d7-8180186dbdb3)

In SquareDesk 1.0.9 Dark Mode:
![image](https://github.com/user-attachments/assets/3a1c9a50-6d88-4be3-99ee-ffa359e25cc7)

**Comment by mpogue2** [comment]
Oh wait...

I bet your have your preferences set to "Music Names: Song name - Label (e.g. 'Chicken Plucker - RB 123')", whereas I have mine set to "Music Names: Best Guess (or a mixture of both formats)".  

When I tell SquareDesk NOT to use Best Guess, then both of your examples fail for me, as shown:

![image](https://github.com/user-attachments/assets/df170b22-6177-4f53-b7a9-8b52d8bf9313)

**Comment by mpogue2** [comment]
OK, I have a fix for this, but in looking for other labels containing dashes, I found that there aren't any!
In fact, "Circle-D" appears to use "CD" now.

So, the fix that I have works for "4-Bar-B" and "Circle-D" and no other labels containing dashes.
Is this going to work for you?  Do you have any other labels whose names contain dashes (maybe super old label names?)?

Fixed by: 8b5155f0bf715503943a5f7215da4cd24e4f9bec

**Comment by phoef** [comment]
Not in my active repertoire. But "Jay-Bar-Kay".
Thanks!

**Comment by mpogue2** [comment]
Jay-Bar-Kay now handled as part of commit: b0523309b03c08653f1f0f270cacc6d7a339b2bd

**Comment by pkbott** [comment]
> OK, I have a fix for this, but in looking for other labels containing dashes, I found that there aren't any! In fact, "Circle-D" appears to use "CD" now.
> 
> So, the fix that I have works for "4-Bar-B" and "Circle-D" and no other labels containing dashes. Is this going to work for you? Do you have any other labels whose names contain dashes (maybe super old label names?)?
> 
> Fixed by: [8b5155f](https://github.com/mpogue2/SquareDesk/commit/8b5155f0bf715503943a5f7215da4cd24e4f9bec)

Found New Beat records also has dashes in downloaded files but they do something like NB-xxx in theirs.

**Comment by mpogue2** [comment]
Yeah, New Beat is a PITA.  I have files from them that look like:

"One Call Away - NB-412 BGV No Melody.htm"

so, there are TWO dashes, AND the title and the label/label_num are reversed, too!

At some point, SquareDesk just has to give up, and do the best it can (not being a human, it's hard to decode what the music producer's intent was!)....

**Comment by pkbott** [comment]
I get it

**Comment by mpogue2** [comment]
Or maybe I just have to write a bunch of code just for New Beat! :-)

**Comment by pkbott** [comment]
I wouldn’t, it isn’t that big a deal yet it is annoying
